### PR TITLE
[release-7.0] Change server selection timeout to 1s

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Description

* Make server selection a constant
* Do not try to write if no txns being inserted or updated

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

File: src/libServer/LookupServer.cpp [hunk 1]
```diff
@@ -2019,6 +2019,11 @@ Json::Value LookupServer::GetTransactionStatus(const string& txnhash) {
                              "Txn Hash size not appropriate");
     }
 
+    if (!REMOTESTORAGE_DB_ENABLE) {
```
Comment (ansnunez) [inline src/libServer/LookupServer.cpp:2022]: I suggest to make this the very first check, and also change the message to "API not supported" (remote storage is an internal detail)

**Review by ansnunez — CHANGES_REQUESTED** [review]

File: constants.xml [hunk 2]
```diff
@@ -40,6 +40,7 @@
         <REMOTESTORAGE_DB_PORT>27017</REMOTESTORAGE_DB_PORT>
         <REMOTESTORAGE_DB_ENABLE>false</REMOTESTORAGE_DB_ENABLE>
         <REMOTESTORAGE_DB_TLS_FILE/>
+        <REMOTESTORAGE_DB_SERVER_SELECTION_TIMEOUT_MS>1000</REMOTESTORAGE_DB_SERVER_SELECTION_TIMEOUT_MS>
```
[the remark this answers is not in the document]
Comment (KaustubhShamshery) [reply constants.xml:43]: I think the write operation is called atleast 3 times by lookup in an epoch. 5 seconds * 3 would be too much if mongo server is not available

**Review by ansnunez — APPROVED** [review]

File: tests/RemoteStorageDB/CMakeLists.txt [hunk 3]
```diff
@@ -3,5 +3,5 @@ link_directories(${CMAKE_BINARY_DIR}/lib)
 
 add_executable(Test_Mongo Test_mongo.cpp)
 target_include_directories(Test_Mongo PUBLIC ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/tests)
-target_link_libraries(Test_Mongo PUBLIC RemoteStorageDB TestUtils)
+target_link_libraries(Test_Mongo PUBLIC RemoteStorageDB TestUtils AccountData)
```
Comment (kaikawaliu) [inline tests/RemoteStorageDB/CMakeLists.txt:6]: seems not needed

Comment (KaustubhShamshery) [reply tests/RemoteStorageDB/CMakeLists.txt:6]: If you try and build test_mongo it is needed as account data is not linked.

Comment (kaikawaliu) [reply tests/RemoteStorageDB/CMakeLists.txt:6]: okay

**Review by kaikawaliu — APPROVED** [review]

**Pull request merged**